Source/Core/Common/EventHook.h

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "Common/Logging/Log.h"
+#include "Common/StringLiteral.h"
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <vector>
+
+// A hookable event system.
+
+// Define Events in a header as:
+//
+//     using MyLoveyEvent = Event<"My lovely event", std::string>;
+//
+// Register listeners anywhere you need them as:
+//    EventHook myHook = MyLoveyEvent::Register([](std::string foo) {
+//       // Do something
+//   }, "Name of the hook");
+//
+// The hook will be automatically unregistered when the EventHook object goes out of scope.
+// Trigger events by doing:
+//
+//   MyLoveyEvent::Trigger("Hello world");
+//
+
+struct HookBase
```

Comment:
[Rule of five](https://en.cppreference.com/w/cpp/language/rule_of_three)

Reply:
I've deleted the copy constructor to make it clear that HookBase shouldn't be copied/moved.